File: app/src/main/java/com/likhit/variants/data/models/Variants.java
package com.likhit.variants.data.models;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;
import java.util.List;

public class Variants implements Serializable {

    @SerializedName("variant_groups")
    private List<VariantGroup> variantGroups;

    @SerializedName("exclude_list")
    private List<List<ExcludeItem>> excludeList;

    public Variants(Variants variants) {
        this.variantGroups = variants.getVariantGroups();
        this.excludeList = variants.getExcludeList();
    }

    public Variants(List<VariantGroup> variantGroups) {
        this.variantGroups = variantGroups;
    }

    public Variants(List<VariantGroup> variantGroups, List<List<ExcludeItem>> excludeList) {
        this.variantGroups = variantGroups;
        this.excludeList = excludeList;
    }

    public List<VariantGroup> getVariantGroups() {
        return variantGroups;
    }

    public void setVariantGroups(List<VariantGroup> variantGroups) {
        this.variantGroups = variantGroups;
    }

    public List<List<ExcludeItem>> getExcludeList() {
        return excludeList;
    }
}

File: app/src/main/java/com/likhit/variants/data/models/Variants.kt
package com.likhit.variants.data.models

import com.google.gson.annotations.SerializedName

import java.io.Serializable

class Variants : Serializable {

    @SerializedName("variant_groups")
    var variantGroups: List<VariantGroup>? = null

    @SerializedName("exclude_list")
    var excludeList: List<List<ExcludeItem>>? = null
        private set

    constructor(variants: Variants) {
        this.variantGroups = variants.variantGroups
        this.excludeList = variants.excludeList
    }

    constructor(variantGroups: List<VariantGroup>) {
        this.variantGroups = variantGroups
    }

    constructor(variantGroups: List<VariantGroup>, excludeList: List<List<ExcludeItem>>) {
        this.variantGroups = variantGroups
        this.excludeList = excludeList
    }
}
